apps/meteor/server/models/raw/Users.js

```diff
@@ -2875,4 +2875,27 @@ export class UsersRaw extends BaseRaw {
 	countRoomMembers(roomId) {
 		return this.col.countDocuments({ __rooms: roomId, active: true });
 	}
+
+	findAllOnlineAgentsExcludingManagersAndMonitors(excludeIds = [], includeIds = [], options = {}) {
+		const extraQuery = { _id: { $nin: excludeIds } };
+		if (includeIds.length > 0) {
+			extraQuery._id = {
+				...extraQuery._id,
+				$in: includeIds,
+			};
+		}
+
+		return this.find(
+			{
+				roles: {
+					$all: ['livechat-agent'],
+					$nin: ['livechat-manager', 'livechat-monitor'],
+				},
+				statusLivechat: 'available',
+				status: 'online',
```

Comment:
Shouldn't we also respect `Livechat_enabled_when_agent_idle` setting over here?

Reply:
Hmmmm, we can remove this filter as all we need is the "available" thing